.pre-commit-config.yaml

```
@@ -40,3 +40,12 @@ repos:
           - pytest
           # Zarr v2
           - types-redis
+  - repo: https://github.com/scientific-python/cookie
+    rev: 2024.04.23
+    hooks:
+      - id: sp-repo-review
```

```suggestion
  - repo: https://github.com/scientific-python/repo-review
    rev: 0.11
    hooks:
      - id: repo-review
```
I think better to use the repo-review repository itself?

---

I understand sp-repo-review/cookie is the tool evaluate repos against the guidelines in the Scientific Python Library Development Guide. The repo-review framework has sp-repo-review and validate-pyproject plyugins.

The [repo-review documentation](https://repo-review.readthedocs.io/en/latest/intro.html#pre-commit) suggests we should directly use the plugin we're interested in:
> ### Pre-commit
> You can also use this from pre-commit:
> ```yaml
> - repo: https://github.com/scientific-python/repo-review
>   rev: <version>
>   hooks:
>     - id: repo-review
>       additional_dependencies: ["repo-review[cli]", "sp-repo-review==<version>"]
> ```
> (Insert the current version above, and ideally pin the plugin version, as long as you have a way to auto-update it.)
> 
> Though check your favorite plugin, which might directly support running from pre-commit, and then pre-commit’s pinning system will pin on your plugin, rather than the framework (repo-review).

---

In that example they use https://github.com/scientific-python/repo-review instead of https://github.com/scientific-python/cookie, so I think it's best to change it from `cookie` to `repo-review`?

---

It depends on who applied the relevant commit. Let me investigate.

---

It goes on with:
> Though check your favorite plugin, which might directly support running from pre-commit, and then pre-commit’s pinning system will pin on your plugin, rather than the framework (repo-review).